### ISSUE TITLE
Can't import legacy

### ISSUE THREAD

**Issue by variant22** [EliteMasterEric/Teyvat.moe#94]
react-dom.production.min.js:101 Uncaught DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.

Please try using my markers ["0_16","0_17","0_23","0_30","0_31","0_34","0_39","0_40","0_41","0_42","0_48","0_49","0_50","0_51","0_53","0_58","0_59","0_60","0_61","26_123","26_124","26_127","26_151","26_155","26_158","26_159","26_160","26_161","26_162","26_185","26_187","26_193","26_200","26_201","26_202","26_206","26_207","26_208","26_209","26_210","26_211","26_212","26_213","26_214","26_215","26_216","26_217","26_218","26_219","26_220","26_221","26_222","26_223","26_224","26_225","26_226","26_228","26_229","26_243","26_244","26_245","26_246","26_247","26_248","26_249","26_250","26_251","26_252","26_253","26_254","26_255","26_256","26_257","26_258","26_259","26_260","26_261","26_262","26_263","26_264","26_265","26_266","26_267","26_269","26_270","26_271","26_272","26_275","26_280","26_282","26_283","26_286","26_291","26_292","26_293","26_294","26_296","26_299","26_300","26_309","26_311","26_312","26_315","26_327","26_328","26_329","26_330","26_331","26_332","26_333","26_334","26_335","26_336","26_337","26_338","26_339","26_340","26_341","26_342","26_343","26_344","26_345","26_346","26_347","26_348","26_349","26_350","26_351","26_352","26_353","26_355","26_358","26_360","26_364","26_369","26_374","26_375","26_377","26_378","26_389","26_390","26_391","26_392","26_393","26_398","26_409","26_410","26_411","26_412","26_413","26_414","26_415","26_416","26_417","26_418","26_419","26_420","26_421","26_422","26_423","26_437","26_438","26_439","26_440","26_441","26_442","26_473","26_478","26_481","26_482","26_484","26_500","26_502","26_503","26_514","26_523","26_525","26_527","26_528"]

**Comment by EliteMasterEric** [comment]
Hey,

Are you using the Import Data option or the Import Legacy Data option? You need to use the second one for importing legacy data to work.

**Comment by xtJelly** [comment]
Same issue, confirm button doesn't do anything for both import and legacy import. Tried 2 browsers, one with no addons.

**Comment by xtJelly** [comment]
Also legacy export is always giving out " [] "

**Comment by Cruffssant** [comment]
Same thing happen to me. I follow guide in [here](https://github.com/GenshinMap/genshinmap.github.io/wiki/Frequently-Asked-Questions#how-do-i-transfer-my-data-from-the-old-version-of-genshinmap) and using Import Legacy Data option, not work to me.
I tried click Confirm after I paste my Legacy Data and it doesn't do anything.
I also tried that using two different browser. Chrome and Microsoft Edge.

**Comment by ilhamhe** [comment]
> Hey,
> 
> Are you using the Import Data option or the Import Legacy Data option? You need to use the second one for importing legacy data to work.

I use the export and import legacy options but the confirm button still doesn't do anything either.

**Comment by YuujiNana** [comment]
Can concur here as well. Import legacy confirm button do not work. Would really suck if everything is lost and having to try to go to each location and try to remember if chest was open or still hidden.

**Comment by EliteMasterEric** [comment]
Hey, I've determined the source of the issues with both `Export Legacy Data` and `Import Legacy Data` and am developing a fix.

In the meantime, do not be scared about data loss. The new version of the site uses as separate Local Storage key from the old one, and never clears all keys in local storage (the `Clear Data` button simply resets the one key the site uses). This means that the old data is always present. This can be seen by opening your browser console (`F12` works on both Firefox and Chrome), navigating to the `Console` tab, and typing:

`window.localStorage`

This will output a dropdown list of all the local storage keys in the browser.

All the data for the new site is stored in `genshinmap-preferences`. In the event of a failed data migration or import, the old data will be stored in `genshinmap-preferences-recovery-#` where # is an incrementing value. The old data is stored in separate keys (the status of Anemoculus 26 is stored in key `0_26` for example) as opposed to a single one, but the app never overrides these.

The only way you will lose your data is if you go into your History and clear your browser's storage.

TL;dr Your data isn't gone and I'm making a fix now.

**Comment by EliteMasterEric** [comment]
v0.8.2 should fix this issue, let me know how it goes.

**Comment by ilhamhe** [comment]
Still doesn't work for me. :( I'm just gonna wait for the others to confirm this.

**Comment by ilhamhe** [comment]
Still doesn't work for me. :( I'm just gonna wait for the others to confirm this. 

Btw, this is the error message on the console when I click the confirm button.
![image](https://user-images.githubusercontent.com/37733584/101125877-dd0cda00-362c-11eb-89b0-60e10ccd4e44.png)

**Comment by variant22** [comment]
Still doesn't work. In the old website, there was an Export functionality which when u click it, it'll generate a JSON file. I'm copy-pasting the JSON there. And nothing happened. I'm using Import Legacy. I cleared my browser's storage before, if that might be the root cause.

**Comment by variant22** [comment]
Okay i can confirm that now i can import it successfully. Thank You @MasterEric !

**Comment by ilhamhe** [comment]
I got the other error message when I tried to export+import legacy data again. 
![image](https://user-images.githubusercontent.com/37733584/101128871-1c3e2980-3633-11eb-950c-c7e62cafc39e.png)

here my JSON data:
["27_934","27_2","27_161","27_163","27_190","27_235","27_261","27_851","27_615","27_168","27_690","27_150","27_798","27_899","27_682","27_752","27_170","27_828","27_228","27_210","27_818","27_674","27_84","27_621","27_226","27_1005","27_213","27_129","27_1077","27_203","27_151","27_1004","27_157","27_180","27_234","27_230","27_634","27_5","27_695","27_82","27_158","27_193","27_199","27_859","27_198","27_790","27_676","27_160","27_836","27_796","27_6","27_763","27_942","27_734","27_196","27_223","27_165","27_648","27_133","27_686","27_130","27_1075","27_701","27_884","27_1037","27_137","27_685","27_898","27_147","27_846","27_1109","27_177","27_131","27_219","27_876","27_903","27_257","27_9","27_802","27_629","27_939","27_135","27_622","27_807","27_667","27_125","27_610","27_874","27_765","27_1070","27_156","27_946","27_260","27_1071","27_804","27_882","27_111","27_252","27_211","27_120","27_1040","27_897","27_937","27_3","27_675","27_166","27_689","27_1003","27_812","27_999","27_46","27_195","27_813","27_231","27_783","27_900","27_256","27_253","27_186","27_677","27_683","27_801","27_171","27_697","27_172","27_793","27_136","27_215","27_623","27_816","27_1072","27_932","27_827","27_1135","27_159","27_611","27_855","27_206","27_4","27_169","27_612","27_1035","27_680","27_217","27_255","27_696","27_809","27_167","27_630","27_848","27_144","27_679","27_617","27_764","test","27_263","27_797","27_187","27_141","27_184","27_1063","27_699","27_1030","27_687","27_214","27_194","27_627","27_124","27_880","27_691","27_700","27_1114","26_159","27_1034","27_1064","27_453","27_233","27_833","27_1136","27_795","27_692","27_789","27_202","27_672","27_873","27_250","27_810","27_140","27_53","27_1076","27_1000","27_227","27_944","27_145","27_173","27_745","27_684","27_845","27_794","27_1031","27_1066","27_126","27_885","27_1","27_1068","27_875","27_264","27_860","27_183","27_681","27_247","27_698","27_824","27_614","27_197","27_152","27_847","27_209","27_201","27_805","27_694","27_940","27_782","27_207","27_123","27_146","27_947","27_1028","27_933","27_792","27_829","27_693","27_127","27_878","27_762","27_619","27_1065","27_1029","27_817","27_222","27_883","27_7","27_742","27_808","27_178","27_837","27_200","27_647","27_788","27_221","27_185","27_945","27_259","27_943","27_132","27_162","27_224","27_1073","27_785","27_265","27_814","27_948","27_744","27_251","27_143","27_666","27_149","27_935","27_688","27_138","27_753","27_175","27_857","27_669","27_121","27_258","27_955","27_815","27_735","27_811","27_1138","27_936","27_938","27_225","27_208","27_803","27_212","27_266","27_743","27_153","27_834","27_877","27_81","27_1067","27_673","27_70","27_702","27_1139","27_850","27_204","27_122","27_218","27_1026","27_671","27_825","27_182","27_155","27_838","27_1032","27_954","27_191","optc","27_951","27_192","27_787","27_134","27_174","27_1069","27_872","27_670","27_254","27_800","27_216","27_119","27_613","27_835","27_164","27_229","27_181","27_881","27_1001","27_784","27_128","27_618","27_139","27_861","27_220","27_189","27_941"]

**Comment by variant22** [comment]
@ilhamhe u can try to clear the local database from ur browser setting first.
Should be here chrome://settings/cookies/detail?site=genshinmap.github.io

**Comment by ilhamhe** [comment]
> @ilhamhe u can try to clear the local database from ur browser setting first.
> Should be here chrome://settings/cookies/detail?site=genshinmap.github.io

Hello, @variant22 I think the problem is with my exported JSON. I tried your JSON and works fine. 
I found some texts in my exported JSON data such as "test" and "optc" (it should be numbers like "27_1040"). I tried to remove those and now it works. So, my problem could be another issue.

**Comment by YuujiNana** [comment]
Was able to import to the new version. But seems some locations was lost, mostly in Stormterror's Lair. Some might be new location in other region that wasn't in the old version. No huge problem as easier with just small error then everything lost. 
Thanks for the fix @MasterEric 

*Might be off topic but I can't seems to toggle anything complete after I imported my old data. Clicking on chest show no toggle. And I can't click at all on Trials and Seelie, haven't tried anything else yet. This might be why some things are not marked complete. 

Edit: Summery show them marked but not on the map or can even toggle them.

**Comment by EliteMasterEric** [comment]
Hey,
Toggling the marker as complete requires double clicking the marker. This
is specified in the Help tab, which is the first tab you see when starting
the app.

As for Trials and Seelies, no pop-up appears for a marker if there is no
comment or media entered for that marker.

**Comment by EliteMasterEric** [comment]
This issue has hopefully been resolved in v0.8.3. If no other users report problems I will close the issue.

**Comment by EliteMasterEric** [comment]
Stale. Please open a new issue if you have problems.
